### ISSUE TITLE
Weird behaviour when using fish with tide prompt in Intellij Idea

### ISSUE THREAD

**Issue by clutwo** [IlanCosman/tide#495]
#### Describe the bug

When using fish with tide prompt in intellij idea terminal, every keystroke forces a new line, resulting in text all over the place.
See the attached video.

Fish shell works normally without any custom prompt as well as with starship. As soon as tide prompt is installed, this starts happening.

I have tried each checkbox set and unset in Intellij terminal settings, the result is always the same.

#### Steps to reproduce
Install fish. 
Install tide prompt.
Set terminal to fish in Intellij Settings.

#### Screenshots

[Screencast_20240306_104020.webm](https://github.com/IlanCosman/tide/assets/2114673/1208d248-f6e7-47ec-8533-9825a8d48806)

#### Environment

Output of `tide bug-report`: 

```
fish version: 3.7.0
tide version: 6.1.1
term: xterm-256color
os: Manjaro / KDE
terminal emulator: Intellij
fish startup: 37.63 millis
fisher plugins: patrickf1/fzf.fish jorgebucaran/autopair.fish joseluisq/gitnow kidonng/zoxide.fish gazorby/fish-exa ilancosman/tide@v6

```

Intellij: 2023.3.4 (also happens in 2024.4.1 EAP)

#### Additional context

Not sure

**Comment by PerchunPak** [comment]
I have the same issue with Python version and PyCharm, WebStorm. This problem disappears if I modify sources and delete the part that prints Python version.

**Comment by jimkoen** [comment]
Same issue with WebStorm:

```
fish version: 3.7.0
tide version: 6.1.1
term: xterm-256color
os: macOS Monterey 12.7.2
terminal emulator: IntelliJ Idea IDE family inbuilt
fish startup: 40.58 millis
fisher plugins: jorgebucaran/fisher ilancosman/tide@v6 jorgebucaran/nvm.fish

```

**Comment by clutwo** [comment]
> Probably a dupe of #476

Indeed, didn't think to search for other Jetbrains products besides IDEA before posting.

**Comment by clutwo** [comment]
So after some testing, in my case "kubectl" seems to be the culprit. When removed, it works perfectly find.
